sympy/polys/numberfields.py

```diff
@@ -550,18 +550,21 @@ def _minpoly_compose(ex, x, dom):
         r = sift(f.items(), lambda itx: itx[0].is_Rational and itx[1].is_Rational)
         if r[True] and dom == QQ:
             ex1 = Mul(*[bx**ex for bx, ex in r[False] + r[None]])
-            r1 = r[True]
-            dens = [y.q for _, y in r1]
+            r1 = dict(r[True])
+            dens = [y.q for y in r1.values()]
```

Comment:
Alternatively, if you just want the 2nd value in all the tuples, `dens = [y.q for y in zip(*r[True])]` but when I tested this I was surprised to see that it takes longer to do this than to get the values from the dictionary when the keys are ints. But when the keys are strings (or more complicated objects?) then the zip is faster.  But neither is faster than the simple list comprehension which doesn't have to build anything new, just unpack it.

So I would recommend against changing those lines.

```python
from time import time
a=[(str(i),i+1) for i in range(100000)]

t=time()
b=list(dict(a).values())
print(time()-t)

t=time()
b=list(zip(*a))
print(time()-t)

t=time()
b=[j for i,j in a]
print(time()-t)
...
0.031001806259155273 #zip
0.07300424575805664 #dict
0.008000373840332031 #comprehension
```

Reply:
The reason for choosing `dict` was the ability to easily find and remove the power of -1. I think that the time spent in the construction of `dens` is insignificant as there are typically only few factors. But if it would turn out be significant, it is possible to start with `r1` a list, then extract `dens`, and finally convert `t1` to a `dict`.

Reply:
That makes sense. I should have seen that this was being used for that purpose.